iree/compiler/Dialect/HAL/Target/LLVM/AOT/internal/UnixLinkerTool.cpp

```diff
@@ -31,7 +31,43 @@ class UnixLinkerTool : public LinkerTool {
 
   std::string getToolPath() const override {
     auto toolPath = LinkerTool::getToolPath();
-    return toolPath.empty() ? "ld.lld" : toolPath;
+    if (!toolPath.empty()) return toolPath;
+    if (targetTriple.isAndroid()) {
+// TODO(ataei, benvanik): Windows cross-linking android NDK support.
+#if defined(__unix__) && defined(__x86_64__)
+#if defined(__linux__)
+      return llvm::Twine(std::getenv("ANDROID_NDK"))
+          .concat("/toolchains/llvm/prebuilt/linux-x86_64/bin/")
```

Comment:
I'd just duplicate the code between `#if` branches rather than have the dangling `.concat` line below. Much easier to follow that way.